### PR TITLE
checklist

### PR DESCRIPTION
**Pull request by talal**
Doneness:
- [x] basic functionality: get/list/put operations for all hierarchies
- [x] `limesctl project sync`
- [x] filter domain/project operations by cluster
- [x] automated tests

FYI @reimannf

### CODE REVIEW

**Comment by talal** [comment]
Changes have been implemented as per Review except for:
> Listing all projects can be expensive in large domains. Please prefer the approach used by the python-openstackclient for resolving a user input that can be either a name or an ID...

File: main.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+			tmp.Unit = limes.UnitExbibytes
+		default:
+			return fmt.Errorf("acceptable units: ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB'], got '%s'", rxMatchedList[2])
+		}
+	}
+
+	if _, exists := (*q)[srv]; !exists {
+		(*q)[srv] = make([]cli.Resource, 0)
+	}
+	(*q)[srv] = append((*q)[srv], tmp)
```
Comment (majewsky) [inline main.go]: You don't need the `if !exists` part. `append()` will happily accept `nil` as its first argument and, in that case, allocate a completely fresh slice.

File: pkg/cli/find.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			return nil, errors.New("more than one domain exists with the name " + userInput)
+		}
+
+		for _, dInList := range dList {
+			d.ID = dInList.ID
+			d.Name = dInList.Name
+		}
+	}
+
+	return d, nil
```
Comment (majewsky) [inline pkg/cli/find.go]: As far as I can see, you need another case here (and also in `FindProject`) for when the `userInput` does not refer to any existing domain, i.e. when the listing returns zero entries. Right now, you would be returning an uninitialized (i.e. all-zeroed) Domain in that case.

File: pkg/cli/interface.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	t.set(q)
+}
+
+// RunSyncTask schedules a sync job that pulls quota and usage data for a project from
+// the backing services into Limes' local database.
+func RunSyncTask(p *Project) {
+	_, limesV1 := getServiceClients()
+
+	err := projects.Sync(limesV1, p.DomainID, p.ID, projects.SyncOpts{})
+	handleError("could not sync project: %v", err)
```
Comment (majewsky) [inline pkg/cli/interface.go]: `%v` is not recognized by `handleError`. I guess that's just a leftover from a previous iteration?

**Pull request merged**